bpf/lib/nodeport.h

```diff
@@ -1710,7 +1710,21 @@ static __always_inline int nodeport_snat_fwd_ipv4(struct __ctx_buff *ctx,
 	if (IS_ERR(ret))
 		goto out;
 
+#if defined(ENABLE_EGRESS_GATEWAY_COMMON) && defined(IS_BPF_HOST)
+	if (target.egress_gateway) {
+		/* Send packet to the correct egress interface, and SNAT it there. */
+		ret = egress_gw_fib_lookup_and_redirect(ctx, target.addr,
+							tuple.daddr, ext_err);
+		if (ret != CTX_ACT_OK)
+			return ret;
+
+		if (!revalidate_data(ctx, &data, &data_end, &ip4))
```

Comment:
I'm not sure I follow why this is needed, what's the case for `fib_do_redirect` to both mangle the packet and return `CTX_ACT_OK`?

Reply:
Me neither, but even latest-greatest kernel needs it. It feels like like an over-zealous verifier.

Reply:
Looks like we're doing the same in [bpf_overlay](https://github.com/cilium/cilium/blob/6c31ccfbc2dc20db81e426a5b9a911cfbd39b7ec/bpf/bpf_overlay.c#L438).